packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -59,6 +60,10 @@ export function FocusScope(props: FocusScopeProps) {
     scopeRef.current = nodes;
   }, [children]);
 
+  if (prevContext) {
+    activeScope = scopeRef;
```

Comment:
Not sure this will work correctly in all cases. The FocusScope might be re-rendered while a child scope is active, and take focus away. I think this needs to be done in the focus handler. The active scope needs to not take focus back if the element being focused is in a different scope.

re your earlier comment about order of events, if we attached the focusin/focusout events at the scope level rather than the document level, we could rely on bubbling to handle the order. Inner scopes would receive events first and we could stopPropagation() to prevent them being handled in outer scopes. Also, if events occurred completely outside the scope (e.g. portals), non-active scopes wouldn't even receive events.

We may also need to track not only the active scope but a set of all currently mounted scopes. Then, in the blur event, we could determine if the target was within another scope and ignore it in that case. If the event occurred outside any scope at all (e.g. the body), only then would we move focus back to the active scope.

In this dialog scenario:

1. The child dialog would mount and focus itself.
2. The previous scope would handle the focusout event, and check if the related target is outside any other scope. In this case it's not, so it would do nothing and allow the focus to move to the child scope. Otherwise, it would restore focus to itself.
2. The new focus scope would handle the focusin event on its node, make itself the active focus scope, and stop propagation to any parent scopes.

Let me know if this makes sense. We can chat about it over vc if needed.